django/forms/boundfield.py

```diff
@@ -227,6 +227,22 @@ def build_widget_attrs(self, attrs, widget=None):
             attrs['disabled'] = True
         return attrs
 
+    @property
+    def widget_type(self):
+        """
+        Returns a the class name of the fields widget.
+        This is useful in templates when layout is dependant upon widget type e.g:
+
+        {% for field in form %}
+          {% if field.widget_type == 'checkboxinput' %}
+            # render one way
+          {% else %}
+            # render another way
+          {% endif %}
+        {% endfor %}
+        """
+        return self.field.widget.__class__.__name__.lower()
```

Comment:
We can strip `input` and `widget` (as Claude suggested).

Reply:
Do you mean the widget's `input_type`?

https://github.com/django/django/blob/5fbc0e07a92f192acfa6bc4b09c3732476eeefc6/django/forms/widgets.py#L299

Reply:
No, I was thinking about `'input'` an `'widget'` strings, e.g.
- `MultipleHiddenInput` -> `multiplehidden`,
- `SplitHiddenDateTimeWidget` -> `splithiddendatetime`,
- `PasswordInput` -> `password`.

Does it make sense? :thinking:

Reply:
Ahh, yes. I understand now.